src/networks/invertible_network_conditional_hint_multiscale.jl

```diff
@@ -64,19 +64,20 @@ mutable struct NetworkMultiScaleConditionalHINT <: InvertibleNetwork
     split_scales::Bool
     logdet::Bool
     is_reversed::Bool
+    squeeze_type::String
 end
 
 @Flux.functor NetworkMultiScaleConditionalHINT
 
 # Constructor
 function NetworkMultiScaleConditionalHINT(n_in::Int64, n_hidden::Int64, L::Int64, K::Int64;
-                                          split_scales=false, k1=3, k2=3, p1=1, p2=1, s1=1, s2=1, logdet=true, ndims=2)
+                                          split_scales=false, k1=3, k2=3, p1=1, p2=1, s1=1, s2=1, logdet=true, ndims=2, squeeze_type="normal")
 
     AN_X = Array{ActNorm}(undef, L, K)
     AN_Y = Array{ActNorm}(undef, L, K)
     CL = Array{ConditionalLayerHINT}(undef, L, K)
     if split_scales
-        XY_dims = Array{Tuple}(undef, L-1)
+        XY_dims = fill!(Array{Tuple}(undef, L-1),(1,1))
```

Comment:
@mloubout If I leave this uninitialized then it throws an error when I try to |> gpu. What is a less hacky way of solving this?

Reply:
what error? where are these used again?

Reply:
So on a clean master branch, the multiscale networks with split_scales turned on 
```
using InvertibleNetworks, Flux
using CUDA
####################################################################################################

# Define network
n_in = 2
n_hidden = 64
L = 2
K = 2
# Construct HINT network
CH = NetworkMultiScaleConditionalHINT(n_in, n_hidden, L, K; split_scales=true)|> gpu

#this also gives same error
#CH = NetworkMultiScaleHINT(n_in, n_hidden, L, K; split_scales=true)|> gpu
```

Gives an undefined reference error on that line we are talking about 

```
julia> CH = NetworkMultiScaleHINT(n_in, n_hidden, L, K; split_scales=true)|> gpu
ERROR: UndefRefError: access to undefined reference
Stacktrace:
 [1] getindex at ./array.jl:809 [inlined]
 [2] iterate at ./array.jl:785 [inlined] (repeats 2 times)
 [3] iterate at ./generator.jl:44 [inlined]
 [4] _collect(::Array{Tuple,1}, ::Base.Generator{Array{Tuple,1},Functors.var"#14#15"{IdDict{Any,Any},typeof(cu)}}, ::Base.EltypeUnknown, ::Base.HasShape{1}) at ./array.jl:699
 [5] collect_similar(::Array{Tuple,1}, ::Base.Generator{Array{Tuple,1},Functors.var"#14#15"{IdDict{Any,Any},typeof(cu)}}) at ./array.jl:628
 [6] map(::Function, ::Array{Tuple,1}) at ./abstractarray.jl:2162
 [7] fmap1 at /home/rorozcom3/.julia/packages/Functors/YlETM/src/functor.jl:30 [inlined]
 [8] #fmap#13 at /home/rorozcom3/.julia/packages/Functors/YlETM/src/functor.jl:35 [inlined]
 [9] #14 at /home/rorozcom3/.julia/packages/Functors/YlETM/src/functor.jl:35 [inlined]
 [10] map at ./tuple.jl:160 [inlined] (repeats 3 times)
 [11] map(::Function, ::NamedTuple{(:AN, :CL, :X_dims, :L, :K, :split_scales),Tuple{Array{ActNorm,2},Array{CouplingLayerHINT,2},Array{Tuple,1},Int64,Int64,Bool}}) at ./namedtuple.jl:187
 [12] fmap1(::Function, ::NetworkMultiScaleHINT) at /home/rorozcom3/.julia/packages/Functors/YlETM/src/functor.jl:30
 [13] fmap(::typeof(cu), ::NetworkMultiScaleHINT; cache::IdDict{Any,Any}) at /home/rorozcom3/.julia/packages/Functors/YlETM/src/functor.jl:35
 [14] fmap at /home/rorozcom3/.julia/packages/Functors/YlETM/src/functor.jl:34 [inlined]
 [15] gpu at /home/rorozcom3/.julia/packages/Flux/goUGu/src/functor.jl:74 [inlined]
 [16] |>(::NetworkMultiScaleHINT, ::typeof(gpu)) at ./operators.jl:834
 [17] top-level scope at REPL[9]:1
```

These XY_dims are used to save the dimensions of forward passing data, so that on an inverse pass we can reconstruct its shape back. Split_scale=true makes the output be a 1d vector so we lose the shape of the original data.

Reply:
>These XY_dims are used to save the dimensions o
Then it shouldn't be `Tuple`. Tuples are in most languages and at least on python and julia immutables so anything that is gonna be modified cannot be a tuple. So in this case it should be an Array of Array not an Array of Tuple.

Reply:
Thank you.